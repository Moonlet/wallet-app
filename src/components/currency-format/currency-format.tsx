import React from 'react';
import { Text } from 'react-native';

import styles from './styles';

const formatter = /(\d)(?=(\d{3})+(?!\d))/g;

interface IProps {
    symbol?: string;
    decimals?: string;
    children: any;
}

const CurrencyFormat = (props: IProps) => {
    const value = parseFloat(props.children);
    let output = props.decimals ? value.toFixed(2) : '' + value;
    output = output.replace(formatter, '$1,');

    return <Text style={styles.text}>{output}</Text>;
};

export default CurrencyFormat;
