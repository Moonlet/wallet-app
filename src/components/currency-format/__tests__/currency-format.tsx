import React from 'react';
import { CurrencyFormat } from '../currency-format';
import { shallow } from 'enzyme';

export default describe('CurrencyFormat', () => {
    test('renders corectly', () => {
        expect(shallow(<CurrencyFormat>123.123</CurrencyFormat>).debug()).toMatchSnapshot();
        expect(
            shallow(<CurrencyFormat decimals="2">123.123</CurrencyFormat>).debug()
        ).toMatchSnapshot();
    });
});
