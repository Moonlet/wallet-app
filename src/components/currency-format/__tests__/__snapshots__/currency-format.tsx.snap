// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyFormat renders corectly 1`] = `
"<withTheme(TextComponent) style={[undefined]}>
  123.123
</withTheme(TextComponent)>"
`;

exports[`CurrencyFormat renders corectly 2`] = `
"<withTheme(TextComponent) style={[undefined]}>
  123.12
</withTheme(TextComponent)>"
`;
